### DISCUSSION TITLE
Maven dependencies with a name referencing a property of a parent are not identified

### DISCUSSION THREAD

**Discussion by gaeljw** [renovatebot/renovate#21510] [Suggest an Idea]
### Type of discussion.

I just want to chat

### Tell us more.

Using Maven as package manager and having dependencies where the `artifactId` includes a variable from a Maven parent POM are not getting renovated.

For instance, in the parent:
```xml
<project>
  <properties>
    <scala.binary.version>2.12</scala.binary.version>
  </properties>
</project>
```

And the project to be renovated:
```xml
<project>

  <parent>
      <!-- the one above, published on a repo -->
   </parent>

  <dependencies>
    <dependency>
         <groupId>io.cucumber</groupId>
         <artifactId>cucumber-scala_${scala.binary.version}</artifactId>
         <version>8.5.1</version>
         <scope>test</scope>
      </dependency>
  </dependencies>

</project>
```

No upgrades are suggested for `cucumber-scala_2.12` even though many exist: https://mvnrepository.com/artifact/io.cucumber/cucumber-scala_2.12.

-----

Reproduction repository: https://github.com/gaeljw/renovate-maven-bug (with a different parent and dependency as in my case these are private projects).

```xml

  <!-- this parent contains property gpg.useagent  with value "true" -->
  <parent>
      <groupId>org.apache</groupId>
      <artifactId>apache</artifactId>
      <version>25</version>
   </parent>

   ...

      <!-- gpg.useagent comes from the parent solr -->
      <!-- this case doesn't make sense as the parent and this deps are not related at all but highlights the issue -->
      <dependency>
         <groupId>net.java.truecommons</groupId>
         <artifactId>${gpg.useagent}commons-io</artifactId>
         <version>2.4</version>
      </dependency>
```

In logs, we can see that no updates are detected for this dependency with the reason `name-placeholder`:

```json
{
            "datasource": "maven",
            "depName": "net.java.truecommons:${gpg.useagent}commons-io",
            "currentValue": "2.4",
            "fileReplacePosition": 1072,
            "registryUrls": [
              "https://repo.maven.apache.org/maven2"
            ],
            "depType": "compile",
            "skipReason": "name-placeholder",
            "updates": [],
            "packageName": "net.java.truecommons:${gpg.useagent}commons-io"
          }
```

Whereas if not using variables in the name, it works fine.

The reproduction repository also contains other variants that do work as long as the property is in the same file.

**Comment by rarkins** [comment]
The reproduction repo doesn't seem to be doing anything. Per the logs, it was empty when it was installed, so was disabled in the backend. Try pushing a `renovate.json` to trigger a check. Once there is a working reproduction, please create a feature request, titled "Maven: support variable interpolation in artifactId".

**Reply by gaeljw** [reply]
Sure, I rushed a bit creating the issue, sorry for that.

The reproduction repo doesn't reproduce, I will investigate more.

**Reply by gaeljw** [reply]
The case is actually more complex, it's related to a Maven property defined in a parent POM that Renovates very likely doesn't have access to and it would be quite complex to do.

I'll open a feature request to track it though.

**Reply by rarkins** [reply]
Please provide more details here first. No point creating a feature request if we need to close it due to it being impractical or impossible

**Reply by gaeljw** [reply]
I updated the reproduction repo and the original discussion as per the actual issue / feature request.

I'm not sure at which point you are familiar with Maven but the feature request is basically to support variable interpolation with a variable coming from another file (Maven parent POM, published on a Maven repo). I guess that this would mean that Renovate would need some specific Maven knowledge to be able to download the parent POM and read it. Unless Renovate is already able to download other files needed to interpret a manager file.

**Reply by rarkins** [reply]
I don't think we'd want to require external connections in order to extract dependencies because today we don't do that for any reason

**Reply by gaeljw** [reply]
I get it. And to be fair, it's probably not a common use case anyway.

**Comment by gaeljw** [comment]
For the record, duplicated issue and discussion: https://github.com/renovatebot/renovate/issues/14670, https://github.com/renovatebot/renovate/discussions/24875
